junit5/src/main/java/cz/xtf/junit5/listeners/ProjectCreator.java

```diff
@@ -47,7 +47,7 @@ private void setTestExecutionContext(ExtensionContext context) {
 
     @Override
     public void afterAll(ExtensionContext context) {
-        if (JUnitConfig.cleanOpenShift()) {
+        if (JUnitConfig.cleanOpenShift() && OpenShiftConfig.useNamespacePerTestCase()) {
```

Comment:
You know I was against introducing those `if OpenShiftConfig.useNamespacePerTestCase()`.

I do not understand why `if xtf.openshift.namespace.per.testcase=false` project is not deleted. Isn't it changing current behaviour?

Reply:
> Isn't it changing current behaviour?

Yes, it's intentional change and return to behavior in previous versions (XTF 0.26 and older). Deleting namespaces between test cases started to cause issues when `xtf.openshift.namespace.per.testcase=false` as there is no 100% reliable way to check that the namespace is deleted.

Reply:
> Ok, so problem is we were deleting namespaces between testcases...

Yes.

>I agree we could use if xtf.openshift.namespace.per.testcase=true ( or deleteProjectIfUsedNamespacePerTestCase - it is the same).

 I've already updated the PR and moved problematic if statement to `NamespaceManager.deleteProjectIfUsedNamespacePerTestCase()` so usage of `OpenShiftConfig.useNamespacePerTestCase()` is isolated to this class.

Reply:
Ok. LGTM.

I think idea to be able to run one testsuite in own namespace is nice strategy. Seems it is hard to incorporate into existing XTF seamlessly.

Reply:
Ok, resolve this thread.